src/V3EmitJson.cpp

```diff
@@ -0,0 +1,242 @@
+        std::vector<string> classes_slow;
+        std::vector<string> support_fast;
+        std::vector<string> support_slow;
+        std::vector<string> global;
+        for (AstNodeFile* nodep = v3Global.rootp()->filesp(); nodep;
+             nodep = VN_AS(nodep->nextp(), NodeFile)) {
+            const AstCFile* const cfilep = VN_CAST(nodep, CFile);
+            if (cfilep && cfilep->source()) {
+                if (cfilep->support()) {
+                    if (cfilep->slow()) {
+                        support_slow.push_back(cfilep->name());
+                    } else {
+                        support_fast.push_back(cfilep->name());
+                    }
+                } else {
+                    if (cfilep->slow()) {
+                        classes_slow.push_back(cfilep->name());
+                    } else {
+                        classes_fast.push_back(cfilep->name());
+                    }
+                }
+            }
+        }
+
+        global.emplace_back(V3Options::getenvVERILATOR_ROOT() + "/include/verilated.cpp");
+        if (v3Global.dpi()) {  //
+            global.emplace_back(V3Options::getenvVERILATOR_ROOT() + "/include/verilated_dpi.cpp");
+        }
+        if (v3Global.opt.vpi()) {
+            global.emplace_back(V3Options::getenvVERILATOR_ROOT() + "/include/verilated_vpi.cpp");
+        }
+        if (v3Global.opt.savable()) {
+            global.emplace_back(V3Options::getenvVERILATOR_ROOT() + "/include/verilated_save.cpp");
+        }
+        if (v3Global.opt.coverage()) {
+            global.emplace_back(V3Options::getenvVERILATOR_ROOT() + "/include/verilated_cov.cpp");
+        }
+        if (v3Global.opt.trace()) {
+            global.emplace_back(V3Options::getenvVERILATOR_ROOT() + "/include/" + v3Global.opt.traceSourceBase()
+                                + "_c.cpp");
+        }
+        if (v3Global.usesProbDist()) {
+            global.emplace_back(V3Options::getenvVERILATOR_ROOT() + "/include/verilated_probdist.cpp");
+        }
+        if (v3Global.usesTiming()) {
+            global.emplace_back(V3Options::getenvVERILATOR_ROOT() + "/include/verilated_timing.cpp");
+        }
+        if (v3Global.useRandomizeMethods()) {
+            global.emplace_back(V3Options::getenvVERILATOR_ROOT() + "/include/verilated_random.cpp");
+        }
+        global.emplace_back(V3Options::getenvVERILATOR_ROOT() + "/include/verilated_threads.cpp");
+        if (v3Global.opt.usesProfiler()) {
+            global.emplace_back(V3Options::getenvVERILATOR_ROOT() + "/include/verilated_profiler.cpp");
+        }
+        if (!v3Global.opt.libCreate().empty()) {
+            global.emplace_back(v3Global.opt.makeDir() + "/" + v3Global.opt.libCreate() + ".cpp");
+        }
+
+        *of << "\t\"sources\": {\n\t\t\"global\": [\n";
+        source_list(of, global);
+        *of << "\t\t],\n\n";
+
+        *of << "\t\t\"classes_slow\": [\n";
+        source_list(of, classes_slow);
+        *of << "\t\t],\n\n";
```

Comment:
Make a common function to do this, then all these will look much cleaner.

```suggestion
        emitSourceList(of, "classes_slow", classes_slow);
```